src/lib.rs

```diff
@@ -628,6 +641,11 @@ impl<S: io::Read + io::Write> TlsStream<S> {
         Ok(self.0.tls_server_end_point()?)
     }
 
+    /// Returns the negotiated protocols
+    pub fn negotiated_protocols(&self) -> Result<Vec<String>> {
```

Comment:
ALPN only allows a single protocol to be negotiated, so this should return a `Result<Option<String>>`.